helpers/base.py

```diff
@@ -131,6 +144,8 @@ def is_ioc(rec):
             continue
         results = fetch_values_by_datatype(rec, datatype)
         for result in results:
+            if isinstance(result, str):
+                result = result.lower() if lowercase_ioc else result.upper()
```

Comment:
Why are we converting to uppercase?

Reply:
In the use case that user has IOCs are uppercase in the IOC files. `lowercase_ioc` indicates if IOCs in IOC files are lowercase (`lowercase_ioc=True`) or uppercase (`lowercase_ioc=False`).

Reply:
Thanks :) What are your thoughts on performance? e.g.
(a) I think this means we're going to perform a lower() or upper() for every string?
(b) If the file already is lowercased, will we still incur the cost?
(c) what happens if one IOC type needs to be lowercased but another doesn't?
(d) can you share details on your approach? e.g. why this vs. lowercasing all hashes and domains on load

Reply:
Thanks for bringing up performance concern. I should've wrapped it up. 
> (a) I think this means we're going to perform a lower() or upper() for every string?

It will perform a lower() or upper() for every string returned by Data Normalization helper. It means only the normalized log Data will be performed a lower() or upper() method. 

> (b) If the file already is lowercased, will we still incur the cost?
> (c) what happens if one IOC type needs to be lowercased but another doesn't?
> (d) can you share details on your approach? e.g. why this vs. lowercasing all hashes and domains on load

1. The lower() method is applied to log data which would have the different format (lowercase, uppercase or mixed case) from different log sources. 
  We could add a check to see if a string is lowercased using method `islower()`. But after measuring both approaches, there is no advantage to add the additional check. Thanks to @ryandeivert 

2. IOC file can be easily formatted when we fetch and write them to a file in the script. And we should format IOC file in the script. It is costly doing in Lambda function because the user may have millions custom IOC.